src/extract.ts

```diff
@@ -1,15 +1,41 @@
 import { promises as fs } from 'fs';
 import * as github from '@actions/github';
 import { Config, ToolType } from './config';
+import { WebhookPayload } from '@actions/github/lib/interfaces';
 
-export interface BenchmarkResult {
+export type BenchmarkResult = RobloxBenchmarkResult | OtherBenchmarkResult;
+
+export interface BaseBenchmarkResult {
+    type?: unknown;
     name: string;
     value: number;
     range?: string;
     unit: string;
     extra?: string;
 }
 
+export enum RobloxUnit {
```

Comment:
FPS isn't cachegrind-specific, but I wonder if we can make this more generic.

Reply:
I don't understand what your point is here. This is not cachegrind-specific because we also work with fps(ops/sec) and Δt benchmarks.

The idea to limit the accepted units is so that we know exactly if a given benchmark is "the bigger the better" or "the smaller the better". We use that to compare benchmarks correctly in the `write.ts` file

Reply:
Is there a way to make this clearer? Nothing that I'm looking at here in the code is communicating that there's a distinction between bigger preferred and smaller preferred.